### ISSUE TITLE
[Bug] Swiper fails and returns "Args out of range: .."

### ISSUE THREAD

**Issue by ianpan870102** [abo-abo/swiper#2654]
I was working on an .org document earlier today and whenever I invoke swiper it fails and returns:

```text
Args out of range: nil, 23058433009213693951
```
For reference, this is the .org file I've been using: https://github.com/ianpan870102/FYP_stuff/blob/master/README.org

The document isn't particularly long (albeit not short either).

**Comment by basil-conto** [comment]
I can't reproduce the error; here's what I tried:

1. `make plain`
2. <kbd>C-x</kbd><kbd>C-f</kbd>`/path/to/README.org`<kbd>RET</kbd>
3. <kbd>C-s</kbd>`prof`<kbd>RET</kbd>

Can you reproduce the error starting from `emacs -Q` or `make plain`? If so, please provide a detailed account of the steps involved.  Otherwise, please try to narrow down which part(s) of your configuration give rise to the problem.  Either way, when reporting a bug please state your full Emacs, Swiper, and Org versions, otherwise it is much harder to debug issues.

**Comment by ianpan870102** [comment]
> I can't reproduce the error

Upon further investigation, it seems that it happens undeterministically, and I fail to reproduce it every time too. I'm using Emacs 27.0.91, Org-mode 9.3, and Swiper 20200503.

Just by seeing the error message, does it ring a bell?

**Comment by basil-conto** [comment]
> Just by seeing the error message, does it ring a bell?

Not for me, sorry.

**Comment by thornjad** [comment]
I see this issue one or two times per day, but for the life of me I can't find a pattern to reproduce reliably. Luckily, I was able to capture a traceback by running with `debug-on-error` on for a couple days!

```
Debugger entered--Lisp error: (args-out-of-range nil 2305843009213693951)
  vertical-motion((2305843009213693951 . 1))
  line-move-visual(1 t)
  line-move(1 t)
  #f(compiled-function (arg) #<bytecode 0xa5bf68faf811e0>)(1)
  swiper--candidates()
  swiper(nil)
  counsel-grep-or-swiper()
  funcall-interactively(counsel-grep-or-swiper)
  call-interactively(counsel-grep-or-swiper nil nil)
  command-execute(counsel-grep-or-swiper)
```

After this error occurs, if I simply call swiper again, it works as expected again.

**Comment by Atemu** [comment]
I also come across this occasionally. I thought this was an instability with emacsGcc or even just emacsGit but it just happened on emacs-27 too.

If it happens to you and you do not want to restart emacs, just try opening swiper a few more times; that makes it work again for some strange reason.

BTW, you can get the lisp debugger without always having `debug-on-error` on by using `M-: (swiper) RET`.

**Comment by Atemu** [comment]
This issue still occurs. It usually also occurs twice in a row.

This just happened to me again and I was able to consistently make it happen again by moving backwards in evil-mode (`b`) and then opening swiper.

After I enabled debug-on-error, I was able to capture this trace once but then the bug went away again, no matter how far back I go.

```
Debugger entered--Lisp error: (args-out-of-range nil 2305843009213693951)
  line-move-visual(1 t)
  line-move(1 t)
  #<subr F616e6f6e796d6f75732d6c616d626461_anonymous_lambda_23>(1)
  swiper--candidates()
  swiper()
  funcall-interactively(swiper)
  command-execute(swiper)
```

**Comment by thornjad** [comment]
Yeah this error continues to happen for me as well, though it seems less frequent than it did than at the time of my last comment in 2020. This currently occurs for me once a week or even less. I can confirm the note that it tends to happen twice in a row. 

In four years of running into this error, I still haven't noticed a pattern as to when it does or doesn't happen. It does tend to happen more frequently for me in Python than any other language, but not exclusively, but that could also be usage bias.

**Comment by Atemu** [comment]
Is there anything us users could do to collect more meaningful data on this?
